File: Backend/src/main/java/com/healthcare/service/HospitalService.java
package com.healthcare.service;

import java.util.List;

import com.healthcare.dto.HospitalDTO;

public interface HospitalService {

	public List<HospitalDTO> getAllHospitals();

	public HospitalDTO getHospitalById(Long id);

	public HospitalDTO updateHospital(Long id, HospitalDTO updatedHospital);

	public HospitalDTO saveHospital(HospitalDTO hospitalDTO);

	public Boolean deleteHospitalById(Long id);

	public List<HospitalDTO> searchHospitalsByName(String name);
}

File: Frontend/src/app/services/hospital.service.ts
import { Injectable } from "@angular/core";
import { Hospital } from "../models/hospital";

@Injectable({
  providedIn: "root",
})
export class HospitalService {
  constructor() {}

  getAllHospitals(): any {
    return null;
  }

  getHospitalById(id: number): any {
    return null;
  }

  createHospital(hospital: Hospital): any {
    return null;
  }

  updateHospital(id: number, hospital: Hospital): any {
    return null;
  }

  deleteHospital(id: number): any {
    return null;
  }

  searchHospitalsByName(name: string): any {
    return null;
  }
}
